SharedExpensesApp/app/src/main/java/com/example/sharedexpensesapp/ui/screens/BalanceScreen.kt

```diff
@@ -0,0 +1,123 @@
+package com.example.sharedexpensesapp.ui.screens
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.material3.CardDefaults
+import androidx.compose.material3.ElevatedCard
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.example.sharedexpensesapp.R
+import com.example.sharedexpensesapp.datasource.DataSource
+import com.example.sharedexpensesapp.model.Expense
+import com.example.sharedexpensesapp.model.GroupItem
+import com.example.sharedexpensesapp.ui.screens.composables.Dropdown
+
+@Composable
+fun BalanceScreen(
+    modifier: Modifier = Modifier,
+    viewModel: GroupViewModel = viewModel(key = "application")
+) {
+    val groups by viewModel.groups.collectAsState()
+    val selectedGroup by viewModel.selectedGroup.collectAsState()
+    val expensesOfGroup = DataSource.expenses.filter { expense ->
+        expense.groupId == selectedGroup.name
+    }
+
+    Column(modifier = modifier) {
+        Dropdown(
+            selectedItem = selectedGroup.name,
+            items = groups.map { group -> group.name },
+            onValueChange = { viewModel.setSelectedGroup(it) },
+            onItemSelected = { viewModel.setSelectedGroup(it) }
+        )
+        LazyColumn {
+            items(expensesOfGroup) { expense ->
+                ExpenseCard(expense = expense)
+            }
+        }
+
+    }
+}
+
+@Composable
+fun ExpenseCard(expense: Expense) {
```

Comment:
There is confusion here which I will try to clarify. On this screen, we want to display **debts** not **expenses**. **Debt** is the overall amount of money one person owes to another, therefore it has one payer and one receiver (as correctly implemented in class Debt). Also, a description is not needed. An example of debt would be: 
```
Eva owes Bob 120 EUR
```
**Expense** is a single payment which includes one payer, multiple participants, description and date. Screen displaying expenses is already implemented on master branch. An example of expense would be:
```
Eva paid 60 EUR for food for herself, Bob and Alice, on 14-12-2023
```
Ideally, the bottom navigation bar should switch between screen with expenses (Group Page) and screen with debts (Balance Page) in context of one group